app/gui2/src/util/ast/node.ts

```diff
@@ -5,15 +5,30 @@ import { Vec2 } from '@/util/data/vec2'
 export function nodeFromAst(ast: Ast.Ast): Node | undefined {
   const nodeCode = ast instanceof Ast.Documented ? ast.expression : ast
   if (!nodeCode) return
+  const pattern = nodeCode instanceof Ast.Assignment ? nodeCode.pattern : undefined
+  const rootSpan = nodeCode instanceof Ast.Assignment ? nodeCode.expression : nodeCode
   return {
     outerExprId: ast.id,
-    pattern: nodeCode instanceof Ast.Assignment ? nodeCode.pattern : undefined,
-    rootSpan: nodeCode instanceof Ast.Assignment ? nodeCode.expression : nodeCode,
+    pattern,
+    rootSpan,
     position: Vec2.Zero,
     vis: undefined,
+    selfArgumentId: findSelfArgument(rootSpan),
```

Comment:
I'm a little worried about naming this a "self argument", since this term isn't actually very clear. In this context, it is just about the syntactic position of "first element of the access chain of primary application". In context of `WidgetFunction` we use that name differently, since the "self argument" may actually be an expression of a named argument present in the prefix chain (i.e. `self=operator1`), depending on many factors (i.e. if the called function is a static).

What I used to call this specific thing (the "syntactic self") is a "function application subject", as used e.g. in the `subjectInfo` computed value or `subjectAsSelf` field of `CallInfo` type. Since we don't care whether or not this subject is an actual "self argument" of the application, I'd stick to that name here too.